toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Records.jsm

```diff
@@ -0,0 +1,44 @@
+
+{%- for record in ci.iter_record_definitions() %}
```

Comment:
We've tried several ways of iterating over types to generate helper code and I think this one might be the best.

One tricky part to is generating code like `CallbackInterfaceRuntime.jsm`, which we don't want to generate if there's no callback interfaces and if there's any callback interfaces we want to generate it exactly once.  That could be handled very simply by adding a `{% if ci.has_callback_interface() %}` call at the top of the template.